client/signup/steps/survey/survey-step-v2.jsx

```diff
@@ -107,7 +107,7 @@ export default React.createClass( {
 
 	handleOtherWriteIn( e ) {
 		this.setState( {
-			otherWriteIn: e.target.value
+			otherWriteIn: e.target.value.replace( /^(\s|\.|\,|\:|\!|\?)+|(\s|\.|\,|\:|\!|\?)+$/g, '' ),
```

Comment:
>I'm down, but isn't [^\w]|\s redundant? (Shouldn't [^\w] include \s too?)

I think `[^\w\s]` would allow whitespace - "if not alphanumeric or whitespace". ¯\_(ツ)_/¯ regex.

>Also, do we have polyfills in place to allow the u ES6 RegExp flag?

No idea would probably be a good question for someone on the i18n team. /cc @yoavf